app/javascript/angular/code/services/buildGraphOptions.js

```diff
@@ -155,11 +155,18 @@ const buildTooltip = ({
     onMouseOverMultiple
   }),
   positioner: function(labelWidth, labelHeight, point) {
+    const pointerWidth = 10;
     const tooltipX = Math.min(
-      Math.max(0, point.plotX - labelWidth / 2), // (extreme right, middle)
-      this.chart.plotWidth - labelWidth // extreme left
+      Math.max(0, point.plotX - (labelWidth - pointerWidth) / 2), // (extreme left, middle)
+      this.chart.plotWidth - labelWidth // extreme right
     );
-    const tooltipY = point.plotY - labelHeight - 10;
+
+    const pointerHeight = 10;
+    let tooltipY = point.plotY - labelHeight - pointerHeight;
+    if (tooltipY < 0) {
+      // changes tooltip position at the top of the graph
```

Comment:
do you mean that it sets the position at the top of the graph?

Reply:
No, I mean that when the tooltip is close to the top of the graph the position is adjusted to prevent it overlapping with the line on the graph

Reply:
cool, could you maybe include the above in the comment?